server/src/test/java/io/crate/planner/optimizer/rule/RewriteFilterOnCrossJoinToInnerJoinTest.java

```diff
@@ -115,7 +115,7 @@ public void test_rewrite_equ_join_filter_on_top_of_cross_join_to_inner_join_on_s
     }
 
     @Test
-    public void test_rewrite_equ_join_filter_on_top_of_nested_cross_join_to_inner_join() {
+    public void test_do_not_rewrite_when_underlying_sources_not_matching_filter() {
         var join1 = new JoinPlan(t1, t3, JoinType.CROSS, null);
         var join2 = new JoinPlan(t1, join1, JoinType.CROSS, null);
         var filter = new Filter(join2, e.asSymbol("doc.t1.a = doc.t2.b and doc.t1.a = doc.t3.c"));
```

Comment:
looking at this, it seems artifical, since `doc.t2.b` will throw an error, as `t2` doesn't participate in the sources.
Is there an actual query that could lead to such a filter?
If not, imho we should remove this test altogether.

Reply:
But then the improvement is untested.

Reply:
Please bear with me if I'm missing something.
My point is that if we cannot end up to such a situation with a real query, then the prod code is also not needed?

Reply:
This is moved into a seperate pr and unrelated to the problem.